sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -317,6 +319,7 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     val ds = Seq(("a", 10), ("a", 20), ("b", 1), ("b", 2), ("c", 1)).toDS()
     val grouped = ds.groupBy($"_1").keyAs[String]
     val agged = grouped.mapGroups { case (g, iter) => (g, iter.map(_._2).sum) }
+    assert(agged.queryExecution.executedPlan.missingInput.isEmpty)
```

Comment:
Maybe we should just have `checkAnswer` assert that there are no physical or logical nodes that have missing input (after analysis)?